pkg/str/str.go

```diff
@@ -29,3 +29,33 @@ func Permutations(set map[string]struct{}, f func([]string)) {
 
 	perm(list, f, 0)
 }
+
+func permInt(a []int, f func([]int), i int) {
+	if i > len(a) {
+		f(a)
+		return
+	}
+
+	permInt(a, f, i+1)
+
+	for j := i + 1; j < len(a); j++ {
+		a[i], a[j] = a[j], a[i]
+		permInt(a, f, i+1)
+		a[i], a[j] = a[j], a[i]
+	}
+}
+
+// PermutationsInts call f on all possible combination of the set.
+func PermutationsInts(set map[int]struct{}, f func([]int)) {
+	var (
+		list = make([]int, len(set))
+		i    = 0
+	)
+
+	for v := range set {
+		list[i] = v
+		i++
+	}
+
+	permInt(list, f, 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '0', create a named constant for it](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)